controller/src/main/java/org/jboss/as/controller/AttributeParser.java

```diff
@@ -194,6 +194,10 @@ public void parseElement(AttributeDefinition attribute, XMLExtendedStreamReader
                 }
                 ParseUtils.requireNoContent(reader);
             }
+            // the object list element was specified but it is empty, so initialize it with an empty list.
+            if (!operation.get(attribute.getName()).isDefined()) {
```

Comment:
This shouldn't be needed, as when you are resolving attribute with AD#resolveModelAttribute() it should take care of that.
Setting empty values to attributes in general is wrong and redundant.

Looking at how resolving of this attribute is done in https://github.com/wildfly/wildfly/blob/master/security-manager/src/main/java/org/wildfly/extension/security/manager/SecurityManagerSubsystemAdd.java#L113
It is clearly wrong.